File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.js
function solution(numer1, denom1, numer2, denom2) {
    let num = numer1 * denom2 + numer2 * denom1;
    let den = denom1 * denom2;
    let gcd = 0;
    
    for (let i = Math.min(num, den); i >= 1; i--) {
        if (num % i == 0 && den % i == 0) {
            gcd = i;
            break;
        }
    }
    
    return [num / gcd, den / gcd]
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.py
def solution(numer1, denom1, numer2, denom2):
    num = numer1 * denom2 + numer2 * denom1
    den = denom1 * denom2
    gcd = 0
    
    for i in range(min(num, den), 0, -1):
        if num % i == 0 and den % i == 0:
            gcd = i
            break
    
    return [num / gcd, den / gcd]